test/unit/org/apache/cassandra/db/tries/ConsistencyTestBase.java

```diff
@@ -55,39 +58,73 @@ public abstract class ConsistencyTestBase<C, T extends BaseTrie<C, ?, T>, R exte
      * partition level as well as consistency.
      */
     public final Predicate<InMemoryTrie.NodeFeatures<C>> FORCE_COPY_PARTITION = features -> isPartition(features.content());
+    /**
+     * Force copy every modified cell below the partition/enumeration level. Provides atomicity of mutations within the
+     * partition level as well as consistency.
+     */
+    public final Predicate<InMemoryTrie.NodeFeatures<TestRangeState>> FORCE_COPY_PARTITION_RANGE_STATE = features -> isPartition(features.content());
     /**
      * Force copy every modified cell below the earliest branching point. Provides atomicity of mutations at any level,
      * but readers/walkers may see inconsistent views of the data, in the sense that older mutations may be missed
      * while newer ones are returned.
      */
-    public final Predicate<InMemoryTrie.NodeFeatures<C>> FORCE_ATOMIC = features -> features.isBranching();
+    public final static Predicate<InMemoryTrie.NodeFeatures<?>> FORCE_ATOMIC = features -> features.isBranching();
+    public final static <Q> Predicate<InMemoryTrie.NodeFeatures<Q>> forceAtomic()
+    {
+        return (Predicate<InMemoryTrie.NodeFeatures<Q>>) (Predicate<?>) FORCE_ATOMIC;
+    }
     /**
      * Do not do any additional copying beyond what is required to build the tries safely for concurrent readers.
      * Mutations may be partially seen by readers, and older mutations may be missed while newer ones are returned.
      */
-    public final Predicate<InMemoryTrie.NodeFeatures<C>> NO_ATOMICITY = features -> false;
+    public final static Predicate<InMemoryTrie.NodeFeatures<?>> NO_ATOMICITY = features -> false;
+    public final static <Q> Predicate<InMemoryTrie.NodeFeatures<Q>> noAtomicity()
+    {
+        return (Predicate<InMemoryTrie.NodeFeatures<Q>>) (Predicate<?>) NO_ATOMICITY;
+    }
 
     abstract R makeTrie(OpOrder readOrder);
 
     abstract C value(ByteComparable b, ByteComparable cprefix, ByteComparable c, int add, int seqId);
+
     abstract C metadata(ByteComparable b);
+
     abstract String pk(C c);
+
     abstract String ck(C c);
+
     abstract int seq(C c);
+
     abstract int value(C c);
+
     abstract int updateCount(C c);
 
     abstract T makeSingleton(ByteComparable b, C content);
+
     abstract T withRootMetadata(T wrapped, C metadata);
 
     abstract T merge(Collection<T> tries, Trie.CollectionMergeResolver<C> mergeResolver);
+
     abstract void apply(R trie,
                         T mutation,
                         InMemoryBaseTrie.UpsertTransformer<C, C> mergeResolver,
                         Predicate<InMemoryTrie.NodeFeatures<C>> forcedCopyChecker) throws TrieSpaceExhaustedException;
 
+    abstract void delete(R trie,
```

Comment:
Could be named `apply` or `applyRange` to make it more generic.

Reply:
The consistency test does only use this to delete data in the trie (with or without placing a tombstone). The name should reflect this.